test/replaytests/versioned_workflow.go

```diff
@@ -0,0 +1,213 @@
+func VersionedWorkflowV2(ctx workflow.Context, _ string) (string, error) {
+	ctx = workflow.WithActivityOptions(ctx, activityOptions)
+
+	var result string
+	var err error
+
+	version := workflow.GetVersion(ctx, TestChangeID, workflow.DefaultVersion, 1, workflow.ExecuteWithMinVersion())
+	if version == workflow.DefaultVersion {
+		err = workflow.ExecuteActivity(ctx, FooActivityName, "data").Get(ctx, &result)
+	} else {
+		err = workflow.ExecuteActivity(ctx, BarActivityName, "data").Get(ctx, &result)
+	}
+	if err != nil {
+		return "", err
+	}
+
+	return result, nil
+}
+
+// VersionedWorkflowV3 is the third version of the workflow. It supports DefaultVersion and Version 1 as well.
+// However, all workflows started by this version will have the change ID set to Version 1.
+// It supports workflow executions started by VersionedWorkflowV1 and VersionedWorkflowV2,
+// as all of them will have the change ID set to DefaultVersion, and it supports them.
+// It also supports workflow executions started by VersionedWorkflowV3 and VersionedWorkflowV4,
+// because the code supports execution of Version 1 of the workflow.
+func VersionedWorkflowV3(ctx workflow.Context, _ string) (string, error) {
+	ctx = workflow.WithActivityOptions(ctx, activityOptions)
+
+	var result string
+	var err error
+
+	version := workflow.GetVersion(ctx, TestChangeID, workflow.DefaultVersion, 1)
+	if version == workflow.DefaultVersion {
+		err = workflow.ExecuteActivity(ctx, FooActivityName, "data").Get(ctx, &result)
+	} else {
+		err = workflow.ExecuteActivity(ctx, BarActivityName, "data").Get(ctx, &result)
+	}
+	if err != nil {
+		return "", err
+	}
+
+	return result, nil
+}
+
+// VersionedWorkflowV4 is the fourth version of the workflow. It supports only Version 1.
+// All workflows started by this version will have the change ID set to Version 1.
+// It supports workflow executions started by VersionedWorkflowV3 and VersionedWorkflowV4,
+// as all of them will have the change ID set to Version 1.
+func VersionedWorkflowV4(ctx workflow.Context, _ string) (string, error) {
+	ctx = workflow.WithActivityOptions(ctx, activityOptions)
+
+	var result string
+
+	workflow.GetVersion(ctx, TestChangeID, 1, 1)
+	err := workflow.ExecuteActivity(ctx, BarActivityName, "data").Get(ctx, &result)
+	if err != nil {
+		return "", err
+	}
+
+	return result, nil
+}
+
+// VersionedWorkflowV5 is the fifth version of the workflow. It supports Version 1 and 2.
+// All workflows started by this version will have the change ID set to Version 1.
+// It supports workflow executions started by VersionedWorkflowV3, VersionedWorkflowV4,
+// VersionedWorkflowV5, VersionedWorkflowV6
+func VersionedWorkflowV5(ctx workflow.Context, _ string) (string, error) {
+	ctx = workflow.WithActivityOptions(ctx, activityOptions)
+
+	var result string
+	var err error
+
+	version := workflow.GetVersion(ctx, TestChangeID, 1, 2, workflow.ExecuteWithVersion(1))
```

Comment:
no JSONs exist with version 2, which makes v5 / v6 a bit fuzzy on replay_test.go... but the integration tests are probably good enough there